DomainContent/Zaru/applause-app/applause-app.js

```diff
@@ -0,0 +1,407 @@
+//  applause-app.js
+// 
+//  Created by Liv Erickson on 6/4/18
+//  Copyright 2018 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+//  App Icon made by Freepick from www.flaticon.com 
+//
+(function() {
+
+    var INDIVIDUAL_CLAP_URLS = [
+        Script.resolvePath("sounds/Claps/clap-3.wav"),
+        Script.resolvePath("sounds/Claps/clap-4.wav"),
+        Script.resolvePath("sounds/Claps/clap-5.wav"),
+        Script.resolvePath("sounds/Claps/clap-6.wav"),
+        Script.resolvePath("sounds/Claps/clap-7.wav"),
+        Script.resolvePath("sounds/Claps/clap-8.wav"),
+        Script.resolvePath("sounds/Claps/clap-9.wav"),
+        Script.resolvePath("sounds/Claps/clap-10.wav")
+    ];
+
+    var CAN_APPLAUD_SETTING = 'io.highfidelity.applauseEnabled';
+    var SHOULD_DISPLAY_PARTICLES_APPLAUSE = 'io.highfidelity.applauseEnabled.useParticles';
+    var HAS_APPLAUSE_APP_SETTING = 'io.highfidelity.applauseEnabled.appPresent';
+
+    var APPLAUSE_BUTTON_IMAGE = Script.resolvePath('./resources/button.png');
+    var APPLAUSE_BUTTON_PRESSED = Script.resolvePath('./resources/button-pressed.png');
+    var WINDOW_Y_OFFSET = 24;
+    var BUTTON_DIMENSIONS = {x: 221, y: 69};
+    var BUTTON_PRESS_TIMEOUT = 150; //ms
+    var HAND_PROXIMITY_SCALE = 6;
+    var handProximityDistance = MyAvatar.getEyeHeight() / HAND_PROXIMITY_SCALE;
+    var APP_ICON = Script.resolvePath('./resources/hand-icon-by-freepik.png');
+    var APPLAUSE_KEY = 't';
+    var APPLAUSE_VOLUME = 0.25;
+    var HAPTICS = {
+        strength: 0.75,
+        duration: 25,
+        hands: 2
+    };
+
+    var tablet = Tablet.getTablet('com.highfidelity.interface.tablet.system');
```

Comment:
It's possible for this to return `undefined` (easy to happen during times like client startup or script reloading), which will cause all of the methods on `tablet` throughout this script to fail, which will make the script choke.

I've gotten around this by doing things like putting `if (!tablet) { return; }` in functions where `tablet` is invoked. I'd recommend you do the same.

Reply:
Do you do this in onClicked() too? I thought that should only be the `tablet.gotoScreen` calls per the certified app guidelines.

Reply:
Yes, this is what the `wallet.js` `onClicked()` looks like:
```
    function onButtonClicked() {
        if (!tablet) {
            print("Warning in buttonClicked(): 'tablet' undefined!");
            return;
        }
        if (onWalletScreen) {
            // for toolbar-mode: go back to home screen, this will close the window.
            tablet.gotoHomeScreen();
        } else {
            tablet.loadQMLSource(WALLET_QML_SOURCE);
        }
    }
```

I see your point about this not being equivalent to what's in the guidelines 🤔